code/addons/vitest/src/vitest-plugin/index.ts

```diff
@@ -346,6 +349,29 @@ export const storybookTest = async (options?: UserOptions): Promise<Plugin[]> =>
     },
     configureVitest(context) {
       context.vitest.config.coverage.exclude.push('storybook-static');
+
+      if (
+        !core?.disableTelemetry &&
+        !(
+          process.env.STORYBOOK_DISABLE_TELEMETRY &&
```

Comment:
Do we need this? Below we just check `core?.disableTelemetry`?

https://github.com/storybookjs/storybook/pull/31473/files#diff-4ed677d06d97e9f1e7d9ba1524c79011107a374f1a79c601c1595e01501a9876R208

Reply:
Honestly that may be a bug. Reading the other code I think we need to manually check `core.disableTelemetry` and `cliOptions.disableTelemetry` (which looks at the env var) everywhere (which is a pain, for sure).

In the vitest plugin we don't have `cliOptions` so we need to look at the env var ourselves instead.

I think maybe we should refactor the `telemetry()` call to 
  a) just check the env var no matter what (better safe than sorry)
  b) take in `core` and `cliOptions` info

Reply:
Refactor sounds like an improvement!

Reply:
Actually I guess if you set the env variable, you can still override it with the CLI flag, which is why we do it like that at the moment.